File: spec/java/src/io/kaitai/struct/spec/TestStrLiterals.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.StrLiterals;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestStrLiterals extends CommonSpec {
    @Test
    public void testStrLiterals() throws Exception {
        StrLiterals r = StrLiterals.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(
                r.complexStr().toCharArray(),
                new char[] { 0, 1, 2, 7, 8, 10, 13, 9, 11, 12, 27, 61, 7, 10, 36, 9787 }
        );
        assertEquals(
                r.doubleQuotes().toCharArray(),
                new char[] { 34, 34, 34 }
        );
        assertEquals(r.backslashes().toCharArray(),
                new char[] { 92, 92, 92 }
        );
        assertEquals(r.octalEatup().toCharArray(),
                new char[] { 0, 50, 50 }
        );
        assertEquals(r.octalEatup2().toCharArray(),
                new char[] { 2, 50 }
        );
    }
}

File: spec/rust/tests/test_str_literals.rs
use std::fs;
extern crate kaitai;
use self::kaitai::*;
use rust::formats::str_literals::*;

#[test]
fn test_str_literals() -> KResult<()> {
    let bytes = fs::read("../../src/fixed_struct.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let r: OptRc<StrLiterals> = StrLiterals::read_into(&_io, None, None)?;

    assert_eq!(*r.complex_str()?, "\u{0}\u{1}\u{2}\u{7}\u{8}\u{a}\u{d}\u{9}\u{b}\u{c}\u{1b}\u{3d}\u{7}\u{a}\u{24}\u{263b}");
    assert_eq!(*r.double_quotes()?, "\u{22}\u{22}\u{22}");
    assert_eq!(*r.backslashes()?, "\u{5c}\u{5c}\u{5c}");
    assert_eq!(*r.octal_eatup()?, "\u{0}\u{32}\u{32}");
    assert_eq!(*r.octal_eatup2()?, "\u{2}\u{32}");
    Ok(())
}
